### DISCUSSION TITLE
What does Copilot use as context?

### DISCUSSION THREAD

**Discussion by patosgui** [community/community#119697] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hello,

I'm using copilot with vscode to get code completions. I have a couple of files that I absolutely don't want Copilot to use as a context due to licensing. I've found that I can disable Copilot for certain files with [0] and that is enough for me for the time being. I've also seen in [limitations](https://docs.github.com/en/copilot/managing-github-copilot-in-your-organization/configuring-content-exclusions-for-github-copilot#limitations) that:

> Content exclusion prevents Copilot directly accessing the content of excluded files. Copilot may draw information about non-excluded files from semantic information provided by the IDE - for example, type information, or hover-over definitions for symbols used in code. It's possible that the IDE may derive this information from excluded files.

I have a couple of questions:
- If I tell copilot (locally) to ignore certain files, does it work the same way as if I would do it repo/organization wide?
- If I have, for example, an allowed file open in the editor and a forbidden one opened on a tab, can I be sure that copilot won't use it for context?
- (double checking) How does copilot deal with files outside the workspace/repo? does it use them as context if they are in opened in an editor or not at all (assuming that they are allowed files)

To give a little bit more context, I'm using github enterprise and I can't enable path-based restrictions.

Many thanks!

[0]

```
	"github.copilot.enable": {
		"*": false,
		"python": true
	},
```

**Comment by Akash1134** [comment, marked as answer]
Hi @patosgui ,

Thanks for posting in the GitHub Community. 


> If I tell copilot (locally) to ignore certain files, does it work the same way as if I would do it repo/organization wide?


Yes, if you locally configure Copilot to ignore certain files, it should respect these settings and not consider those files when providing suggestions. This would work similarly to setting these exclusions at the repo or organization level. However, the specifics might depend on the exact version and configuration of your Copilot tool.


> If I have, for example, an allowed file open in the editor and a forbidden one opened on a tab, can I be sure that copilot won't use it for context?

Copilot should not use it as context when providing code suggestions, regardless of whether it's open in a tab or not. Copilot should only consider the content of the "allowed" files when making suggestions. 

>  How does copilot deal with files outside the workspace/repo? does it use them as context if they are in opened in an editor or not at all (assuming that they are allowed files)


GitHub Copilot typically uses the context of the file you are currently editing to generate its suggestions, and it's mainly designed to work with files within your current workspace or repository. As for files outside of your workspace or repository, even if they are open in the editor, Copilot does not generally use them as context, unless they are part of the current file's context (e.g., they are imported or referenced in the current file).

Please note that this is based on the general working principles of Copilot and the exact behavior might depend on the specifics of the VS Code Copilot extension, which is a separate entity from this chat assistant.

Hope that helps!

**Comment by patosgui** [comment]
Thank you @Akash1134, that was very helpful!

I'll keep waiting for directory/path based exclusion. Until then, it is great to know what gets used or not :)

**Comment by simpliatanu** [comment]
even if I add more files from current workspace as context it does not use it, it only falls back to current file> why is it so? this is not very useful, often changes are needed in multiple files to solve a problem

**Comment by MrLucifer-92** [comment]
I'll keep waiting for directory/path based exclusion.

**Comment by uperkey** [comment]
Hei! Tässä lyhyet ja selkeät vastaukset:

Paikallinen Copilot-asetus (VS Code) toimii vain sinun koneellasi. Organisaatio-/repo-tason rajoitukset ovat laajempia ja koskevat kaikkia käyttäjiä – ne eivät ole täysin sama asia.

Jos estetty tiedosto on auki välilehdessä, Copilot ei suoraan käytä sitä. Mutta IDE voi silti välittää epäsuoraa tietoa (kuten tyypit), joten 100% varmuutta ei ole.

Työtilan ulkopuolisia tiedostoja Copilot ei yleensä käytä kontekstina, vaikka ne olisivat auki – ellei IDE tuo niistä jotain semanttista tietoa.

Lopuksi: Jos haluat laskea verot helposti Suomessa, käytä [ALV-laskuri](https://alv-laskuri-online.fi/) – se auttaa ALV-veron (arvonlisävero) nopeassa laskemisessa.

**Reply by 2726235824** [reply]
great

**Reply by uperkey** [reply]
thanks mate ! have a good day .

**Comment by uperkey** [comment]
A [Discord Timestamps Generator](https://dctimestampgen.com/) makes it easy to create accurate time and date formats for Discord. Simply select the date and time you need, choose your preferred timestamp style, and copy the generated code directly into Discord. It’s useful for scheduling events, gaming sessions, announcements, meetings, and other activities where people may be in different time zones.
